### PR TITLE
Created Filler Strategy Class

### PR DESCRIPTION
**Pull request by ZakeryFyke**
Created filler_strategy, which iterates over every column of a dataframe and makes a decision on what to do about any missing data in the column, depending on the column type and the proportion of missing data.

### CODE REVIEW

**Comment by ZakeryFyke** [comment]
Ooo, conflicts, my favorite

File: modules/DMZ/data_kit/filler_strategy.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        self.fill_value = -9999
+
+    def get_dataset(self):
+        self.pandas_dataset = pd.read_csv(self.file_name, sep=None)
+
+    def get_missing_ratios_dict(self):
+        missing_ratios = dataset_insight.get_missing_ratios(self.pandas_dataset, "column")
+        self.missing_ratios_dict = dict(zip(self.pandas_dataset.columns.values, missing_ratios))
+
+    def run_fillers(self):
```
Comment (asclines) [inline modules/DMZ/data_kit/filler_strategy.py]: Is this the best way of doing this method? The nested-everything seems wrong

File: modules/DMZ/data_kit/filler_strategy.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        self.run_fillers()
+
+    def get_config_options(self):
+        self.file_name = "titanic_train.csv"
+        self.avg_thresh = .50
+        self.value_thresh = .70
+        self.fill_value = -9999
+
+    def get_dataset(self):
+        self.pandas_dataset = pd.read_csv(self.file_name, sep=None)
```
Comment (asclines) [inline modules/DMZ/data_kit/filler_strategy.py]: There's really no separator?

File: modules/DMZ/data_kit/filler_strategy.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            if self.pandas_dataset[col].dtype != "object":
+                if self.missing_ratios_dict[col] > 0:
+                    if self.missing_ratios_dict[col] <= self.avg_thresh:
+                        self.pandas_dataset[col] = data_filler.fill_missing_data_average(self.pandas_dataset[col])
+                    elif self.missing_ratios_dict[col] <= self.value_thresh:
+                        self.pandas_dataset[col] = data_filler.fill_missing_data(self.pandas_dataset[col], self.fill_value)
+                    else:
+                        self.pandas_dataset.drop(col, axis = 1, inplace = True)
+            # We'd eventually like to run data fillers on all nonstring values and use their filled values to
+            # predict what should fill missing string data, but that's an icebox goal for the time being.
```
Comment (asclines) [inline modules/DMZ/data_kit/filler_strategy.py]: This shouldn't be a comment

File: modules/DMZ/data_kit/filler_strategy.py (hunk 2 above)
Comment (ZakeryFyke) [inline modules/DMZ/data_kit/filler_strategy.py]: If sep is left to none, I believe it autodetects the separator, @ASAAR can correct me if I'm wrong.

Comment (asclines) [inline modules/DMZ/data_kit/filler_strategy.py]: Ahh, right..

File: modules/DMZ/data_kit/filler_strategy.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        self.pandas_dataset = None
+        self.missing_ratios_dict = None
+
+        self.get_config_options()
+        self.get_dataset()
+        self.get_missing_ratios_dict()
+        self.run_fillers()
+
+    def get_config_options(self):
+        self.file_name = "titanic_train.csv"
```
Comment (RyanMcBerg) [inline modules/DMZ/data_kit/filler_strategy.py]: Do we want this to be general, and not specifically the titanic dataset?

Comment (ASAAR) [inline modules/DMZ/data_kit/filler_strategy.py]: Doesn't matter yet, we're waiting on the config tool. There could be a comment next to it with a TODO to fix though.

**Pull request merged**